lib/openstack/openstack_handle/handle.rb

```diff
@@ -28,12 +28,16 @@ class Handle
 
     # Tries both non-SSL and SSL connections to Openstack
     def self.try_connection
-      # attempt to connect without SSL
-      yield "http", {}
-    rescue Excon::Errors::SocketError => err
-      raise unless err.message.include?("end of file reached (EOFError)")
-      # attempt the same connection with SSL
+      # attempt to connect with SSL
       yield "https", {:ssl_verify_peer => false}
+    rescue Excon::Errors::SocketError => err
+      # TODO recognizing something in exception message is not very reliable. But somebody would need to go to excon gem
+      # and do proper exceptions like Excon::Errors::SocketError::UnknownProtocolSSL,
+      # Excon::Errors::SocketError::BadCertificate, etc. all of them inheriting from Excon::Errors::SocketError
```

Comment:
I would like both all Excon::Error::SocketError.subclasses and specific Excon::Error::SocketError::CertificateError.

It's much more stable. If somebody just decides to rephrase the message, to be more understandable, it should not break things. :-)

Yes, excon doesn't have this kind of break down. And they do similar message checks right inside of the Excon::Errors::SocketError. So maybe they will keep that in mind and take messages as part of the code. :-)